File: app/src/main/java/com/canli/oya/traininventoryroom/data/dao/BrandDao.java
package com.canli.oya.traininventoryroom.data.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.canli.oya.traininventoryroom.data.entities.BrandEntry;

import java.util.List;

@Dao
public interface BrandDao {

    @Query("SELECT * FROM brands")
    LiveData<List<BrandEntry>> getAllBrands();

    @Query("SELECT * FROM brands WHERE brandId = :id")
    LiveData<BrandEntry> getChosenBrand(int id);

    @Insert
    void insertBrand(BrandEntry brand);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateBrandInfo(BrandEntry brand);

    @Delete
    void deleteBrand(BrandEntry brand);
}

File: app/src/main/java/com/canli/oya/traininventoryroom/data/dao/BrandDao.kt
package com.canli.oya.traininventoryroom.data.dao

import androidx.lifecycle.LiveData
import androidx.paging.DataSource
import androidx.paging.PagingSource
import androidx.room.Dao
import androidx.room.Query
import com.canli.oya.traininventoryroom.data.BrandEntry
import kotlinx.coroutines.flow.Flow

@Dao
interface BrandDao : BaseDao<BrandEntry> {

    @Query("SELECT * FROM brands ORDER BY brandName")
    fun observeAllPagedBrands(): PagingSource<Int, BrandEntry>

    @Query("SELECT * FROM brands ORDER BY brandName")
    fun observeAllBrands() : Flow<List<BrandEntry>>

    @Query("SELECT * FROM brands ORDER BY brandName")
    suspend fun getBrandList() : List<BrandEntry>

    @Query("SELECT brandName FROM brands ORDER BY brandName")
    suspend fun getBrandNames() : List<String>

    @Query("SELECT * FROM brands WHERE brandId = :id")
    fun observeChosenBrand(id: Int): Flow<BrandEntry>

    @Query("SELECT * FROM brands WHERE brandId = :id")
    suspend fun getChosenBrand(id: Int): BrandEntry
}
